lib/plugins/filter/after_render/external_link.js

```diff
@@ -9,31 +9,31 @@ function externalLinkFilter(data) {
 
   if (!cheerio) cheerio = require('cheerio');
 
-  const $ = cheerio.load(data.content, {decodeEntities: false});
+  const $ = cheerio.load(data, {decodeEntities: false});
   const siteHost = url.parse(config.url).hostname || config.url;
 
-  $('a').each(function() {
+  $('a').each((index, element) => {
     // Exit if the link has target attribute
-    if ($(this).attr('target')) return;
+    if ($(element).attr('target')) return;
 
     // Exit if the href attribute doesn't exists
-    const href = $(this).attr('href');
+    const href = $(element).attr('href');
```

Comment:
I tried insert `console.log('Href: ' + href)` here, I got
```
Href: #Title
Href: #Another-title
```

which is unexpected from
``` js
    const content = [
      '# External link test',
      '1. External link',
      '<a href="https://hexo.io/">Hexo</a>',
      '2. Internal link',
      '<a href="/archives/foo.html">Link</a>',
      '3. Ignore links have "target" attribute',
      '<a href="https://hexo.io/" target="_blank">Hexo</a>',
      '4. Ignore links don\'t have "href" attribute',
      '<a>Anchor</a>',
      '5. Ignore links whose hostname is same as config',
      '<a href="https://example.com">Example Domain</a>'
    ].join('\n');
```

Reply:
Fixed